### PR TITLE
docs: Progress towards ANN001: Missing type annotation for function argument

### PR DESCRIPTION
**Pull request by silasary**
## What type of pull request is this?

- [X] Non-breaking code change
- [ ] Breaking code change
- [ ] Documentation change/addition
- [ ] Tests change

## Description
Adds type hints to a chunk of parameters across the library.  Not enough to turn on the linter yet, but progress is progress, and I don't want to be in merge-conflict hell.

## Changes
- Added types for a fuckton of parameters.
- Made TTLCache generically typed

## Checklist

- [x] I've formatted my code with [Black](https://black.readthedocs.io/en/stable/)
- [x] I've ensured my code works on `Python 3.10.x`
- [x] I've tested my code

### CODE REVIEW

**Comment by LordOfPolls** [comment]
>This PR is not even remotely ready

Suggest converting this to a draft then

**Comment by silasary** [comment]
@LordOfPolls I tried to stay away from things that would conflict with Voice for now, but I make no promises

**Comment by LordOfPolls** [comment]
Ta very much

File: dis_snek/client/utils/cache.py [hunk 1]
```diff
@@ -1,48 +1,53 @@
...
+from typing import Any, Callable, Generic, Iterator, Optional, Tuple, TypeVar
 
 import attrs
 
 from dis_snek.client.utils.attr_utils import define, field
 
 __all__ = ["TTLItem", "TTLCache"]
 
+KT = TypeVar("KT")
+VT = TypeVar("VT")
```
Comment (benwoo1110) [inline dis_snek/client/utils/cache.py:13]: Better to put all the generic `TypeVar` variables in const.py? Since its likely being used in multiple files

**Review by benwoo1110 — COMMENTED** [review]
Generally looks good, ready to merge or still have more type hinting to add?

File: dis_snek/client/utils/cache.py (hunk 1 above)
Comment (silasary) [reply dis_snek/client/utils/cache.py:13]: If I end up having to type more mappings, I'll move them

**Comment by silasary** [comment]
Ready to merge

**Review by benwoo1110 — APPROVED** [review]

**Pull request merged**